tests/components/husqvarna_automower/conftest.py

```diff
@@ -82,20 +83,22 @@ async def setup_credentials(hass: HomeAssistant) -> None:
 @pytest.fixture
 def mock_automower_client() -> Generator[AsyncMock, None, None]:
     """Mock a Husqvarna Automower client."""
-    with patch(
-        "homeassistant.components.husqvarna_automower.AutomowerSession",
-        autospec=True,
-    ) as mock_client:
-        client = mock_client.return_value
-        client.get_status.return_value = mower_list_to_dictionary_dataclass(
-            load_json_value_fixture("mower.json", DOMAIN)
-        )
 
-        async def websocket_connect() -> ClientWebSocketResponse:
-            """Mock listen."""
-            return ClientWebSocketResponse
+    async def websocket_connect() -> ClientWebSocketResponse:
+        """Mock listen."""
+        return ClientWebSocketResponse
+
+    mower_dict = mower_list_to_dictionary_dataclass(
+        load_json_value_fixture("mower.json", DOMAIN)
+    )
 
-        client.auth = AsyncMock(side_effect=websocket_connect)
-        client.commands = AsyncMock()
+    mock = AsyncMock(spec=AutomowerSession)
+    mock.auth = AsyncMock(side_effect=websocket_connect)
```

Comment:
```suggestion
    mock.auth.return_value = ClientWebSocketResponse
```

Should work
If not
```suggestion
    mock.auth.side_effect = websocket_connect
```

Reply:
I just got this working: https://github.com/home-assistant/core/pull/117938/commits/03d4e66d8f03bdcedb4cbacce8ba459d5840c878